charts/s1-agent/templates/_helpers.tpl

```diff
@@ -94,3 +104,29 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{- define "agent.fullname" -}}
+{{- if .Values.agent.fullnameOverride -}}
+{{- .Values.agent.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default "agent" .Values.agent.nameOverride -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "helper.fullname" -}}
+{{- if .Values.agent.fullnameOverride -}}
+{{- .Values.agent.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default "helper" .Values.agent.nameOverride -}}
```

Comment:
do we allow name override? what about the agent code with the hard-coded name? (or it's not from here?)